### PR TITLE
makes the ed209 a child of securitrons instead of a horrifying copypaste

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

does what says on title, becomes a securitron that can shoot people
removes ed209 cell because it doesnt do anything why does it even exist
fixes ed209 crafting using dragnets
sadly, removes lasertag mode from ed209, but ive never seen anyone use it and it made everything a lot more confusing
eds now shoot you even if youre lying down because i guess that you could rest before to avoid getting shot but im not 100% sure

## Why It's Good For The Game

fixes tons of horrifying copypaste
also fixes some minor things

## Changelog
:cl:
del: Lasertag EDs have been removed
code: EDs are now not a horrifying mess of copypasted securitron code
balance: EDs now still shoot you if you're lying down
tweak: you no longer need a battery to make an ed209
fix: fixes ed209 crafting using dragnets instead of disablers
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/secbot.dm [hunk 1]
```diff
@@ -35,6 +35,7 @@
 	var/weaponscheck = FALSE //If true, arrest people for weapons if they lack access
 	var/check_records = TRUE //Does it check security records?
 	var/arrest_type = FALSE //If true, don't handcuff
+	var/ranged = FALSE //used for EDs
```
Comment (nemvar) [inline code/modules/mob/living/simple_animal/bot/secbot.dm]: This is an istype(src) check in disguise.

File: code/modules/mob/living/simple_animal/bot/ed209bot.dm [hunk 2]
```diff
@@ -1,443 +1,72 @@
...
-	var/datum/atom_hud/secsensor = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
-	secsensor.add_hud_to(src)
-
-/mob/living/simple_animal/bot/ed209/turn_on()
-	. = ..()
-	icon_state = "[lasercolor]ed209[on]"
-	mode = BOT_IDLE
-
-/mob/living/simple_animal/bot/ed209/turn_off()
+/mob/living/simple_animal/bot/secbot/ed209/bot_reset()
```
Comment (nemvar) [inline code/modules/mob/living/simple_animal/bot/ed209bot.dm]: This proc does nothing.

File: code/modules/mob/living/simple_animal/bot/ed209bot.dm [hunk 3]
```diff
@@ -1,443 +1,72 @@
...
-
 		var/dst = get_dist(src, C)
 		if(dst <= 1 || dst > 7)
 			continue
-
 		targets += C
 	if(targets.len>0)
 		var/mob/living/carbon/t = pick(targets)
-		if(t.stat != DEAD && (t.mobility_flags & MOBILITY_STAND) && !t.handcuffed) //we don't shoot people who are dead, cuffed or lying down.
+		if(t.stat != DEAD && !t.handcuffed) //we don't shoot people who are dead, cuffed or lying down.
```
Comment (nemvar) [inline code/modules/mob/living/simple_animal/bot/ed209bot.dm]: Undocumented change

File: code/modules/mob/living/simple_animal/bot/ed209bot.dm (hunk 2 above)
Comment (Fikou) [reply code/modules/mob/living/simple_animal/bot/ed209bot.dm]: documented

File: code/modules/mob/living/simple_animal/bot/ed209bot.dm (hunk 3 above)
Comment (Fikou) [reply code/modules/mob/living/simple_animal/bot/ed209bot.dm]: documented

File: code/modules/mob/living/simple_animal/bot/ed209bot.dm (hunk 2 above)
Comment (Fikou) [reply code/modules/mob/living/simple_animal/bot/ed209bot.dm]: but shouldnt it set weapons?

File: code/modules/mob/living/simple_animal/bot/secbot.dm [hunk 4]
```diff
@@ -420,7 +422,10 @@ Auto Patrol: []"},
 			target = C
 			oldtarget_name = C.name
 			speak("Level [threatlevel] infraction alert!")
-			playsound(loc, pick('sound/voice/beepsky/criminal.ogg', 'sound/voice/beepsky/justice.ogg', 'sound/voice/beepsky/freeze.ogg'), 50, FALSE)
+			if(ranged)
+				playsound(src, pick('sound/voice/ed209_20sec.ogg', 'sound/voice/edplaceholder.ogg'), 50, FALSE)
+			else
+				playsound(loc, pick('sound/voice/beepsky/criminal.ogg', 'sound/voice/beepsky/justice.ogg', 'sound/voice/beepsky/freeze.ogg'), 50, FALSE)
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/modules/mob/living/simple_animal/bot/secbot.dm]: good notice, thanks

File: code/modules/mob/living/simple_animal/bot/secbot.dm (hunk 1 above)
Comment (Fikou) [reply code/modules/mob/living/simple_animal/bot/secbot.dm]: is that fine or do i have to replace with istype

**Comment by PKPenguin321** [comment]
excellent commit names

**Comment by cacogen** [comment]
so it can stunlock you?

File: code/modules/mob/living/simple_animal/bot/secbot.dm (hunk 1 above)
Comment (Qustinnus) [reply code/modules/mob/living/simple_animal/bot/secbot.dm]: you use inheritance

**Comment by actioninja** [comment]
stale

**Comment by AnturK** [comment]
Is this done or not ?

**Comment by Fikou** [comment]
> Is this done or not ?

done unless theres bugs

**Pull request merged**